enterprise/backend/src/metabase_enterprise/query_field_validation/api.clj

```diff
@@ -27,27 +27,37 @@
 
 (defn- cards-with-inactive-fields
   [sort-column sort-direction offset limit]
-  (let [additional-joins      (condp = sort-column
+  (let [sort-dir-kw           (keyword sort-direction)
+        additional-joins      (condp = sort-column
                                 "name"           []
-                                "collection"     [[(t2/table-name :model/Collection) :coll] [:= :coll.id :c.collection_id]]
+                                "collection"     [[(t2/table-name :model/Collection) :coll] [:or
+                                                                                             [:= :coll.id :c.collection_id]
+                                                                                             [:= nil :c.collection_id]]]
                                 "created_by"     [[(t2/table-name :model/User) :u] [:= :u.id :c.creator_id]]
                                 "last_edited_at" [])
         order-by-column       (condp = sort-column
-                                "name"           :c.name
-                                "collection"     :coll.name
-                                "created_by"     [:coalesce [:|| :u.first_name " " :u.last_name] :u.first_name :u.last_name :u.email]
-                                "last_edited_at" :c.updated_at)
+                                "name"           [[:c.name sort-dir-kw]]
+                                "collection"     [[[:case [:= nil :coll.name] 0 :else 1] sort-dir-kw]
+                                                  [:coll.name sort-dir-kw]]
+                                "created_by"     [[[:coalesce [:|| :u.first_name " " :u.last_name]
+                                                    :u.first_name :u.last_name :u.email]
+                                                   sort-dir-kw]]
+                                "last_edited_at" [[:c.updated_at sort-dir-kw]])
         extra-selects         (condp = sort-column
-                                "created_by" [:u.first_name :u.last_name :u.email]
-                                [order-by-column])
+                                "name"           [:c.name]
+                                "collection"     [:coll.name]
```

Comment:
It looks like this is the culprit - that some engines want you to select by the full expression and alias to something you use in the order-by:


```
"Order by expression \"CASE WHEN COLL.NAME IS NULL THEN ?1 ELSE ?2 END\" must be in the result list 
```